### PR TITLE
ordering the chart bars

### PR DESCRIPTION
**Pull request by emre2038**
## Description
ordering the chart bars according to the subindicator ordering in their group

## Related Issue
https://trello.com/c/I1G1q6rw/832-chart-does-not-respect-subindicator-group-ordering-est-3hrs

## How to test it locally
* expand rich data panel
* confirm that bars of the charts have the same order as they do in the admin panel

## Screenshots
before : 
image

after : 
image

### Updated
* `profile / indicator.js` to order the data according to the order of `metadata.groups`

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by emre2038** [comment]
[chart does not respect subindicator group ordering [Est: 3hrs]](https://trello.com/c/I1G1q6rw/832-chart-does-not-respect-subindicator-group-ordering-est-3hrs)

**Comment by adieyal** [comment]
@emre2038 code looks good but needs a test

**Comment by adieyal** [comment]
@emre2038 when I test this, the language subindicators disappe
Screenshot from 2021-09-20 15-16-30
ar

File: src/js/profile/blocks/indicator.js [hunk 1]
```diff
@@ -31,4 +34,26 @@ export class Indicator extends ContentBlock {
...
+        let primaryGroup = this.indicator.metadata.primary_group;
+        let groupsOrder = this.indicator.metadata.groups.filter((g) => {
+            return g.name === primaryGroup
+        })[0];
+        let orderedGroups = [];
+
+        groupsOrder.subindicators.forEach((g) => {
+            let d = this.indicator.data.filter((x) => {
+                return x[primaryGroup] === g
+            })[0];
```
Comment (goyal1092) [inline src/js/profile/blocks/indicator.js:48]: Here 0 is returning first of the filtered list.
Fails if there is more than one data row.

File: src/js/profile/blocks/indicator.js [hunk 2]
```diff
@@ -31,4 +34,26 @@ export class Indicator extends ContentBlock {
...
+
+            if(d !== undefined){
+                orderedGroups.push(d);
```
Comment (goyal1092) [inline src/js/profile/blocks/indicator.js:51]: After removing 0 above this line should have been changed to:
`orderedGroups.push(...d);`

**Comment by jbothma** [comment]
Closing in favour of https://github.com/OpenUpSA/wazimap-ng-ui/pull/528 where this is continued

**Pull request closed without merging**